backend/ee/danswer/external_permissions/confluence/doc_sync.py

```diff
@@ -1,19 +1,253 @@
 from typing import Any
 
+from atlassian import Confluence  # type:ignore
 from sqlalchemy.orm import Session
 
+from danswer.access.models import ExternalAccess
+from danswer.connectors.confluence.confluence_utils import (
+    generate_confluence_document_id,
+)
+from danswer.connectors.confluence.rate_limit_handler import (
+    make_confluence_call_handle_rate_limit,
+)
 from danswer.db.models import ConnectorCredentialPair
+from danswer.db.users import batch_add_non_web_user_if_not_exists__no_commit
 from danswer.utils.logger import setup_logger
-from ee.danswer.external_permissions.permission_sync_utils import DocsWithAdditionalInfo
+from ee.danswer.db.document import upsert_document_external_perms__no_commit
+from ee.danswer.external_permissions.confluence.confluence_sync_utils import (
+    build_confluence_client,
+)
 
 
 logger = setup_logger()
 
+_REQUEST_PAGINATION_LIMIT = 100
+
+
+def _get_space_permissions(
+    db_session: Session,
+    confluence_client: Confluence,
+    space_id: str,
+) -> ExternalAccess:
+    get_space_permissions = make_confluence_call_handle_rate_limit(
+        confluence_client.get_space_permissions
+    )
+
+    space_permissions = get_space_permissions(space_id).get("permissions", [])
+    user_emails = set()
+    group_names = set()
+    is_externally_public = False
+    for permission in space_permissions:
+        subjects = permission.get("subjects", {})
+        if "user" in subjects:
+            user_email = subjects["user"]["results"][0]["email"]
+            if user_email:
+                user_emails.add(user_email)
+        elif "group" in subjects:
+            group_name = subjects["group"]["results"][0]["name"]
+            if group_name:
+                group_names.add(group_name)
+        elif (
+            subjects == {}
```

Comment:
This makes no sense to the uninitiated, why does subject need to be an empty dict and those other things like that? Need some comment here